media-gfx/asymptote/patches/asymptote-3.02.patchset

```diff
@@ -0,0 +1,92 @@
+From 7567d1aa9b0bf53cd40306beebbfdb9fd7917e04 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Joachim=20Mairb=C3=B6ck?= <j.mairboeck@gmail.com>
+Date: Fri, 18 Apr 2025 20:59:28 +0200
+Subject: Haiku build fixes
+
+Use the gnu++17 C++ standard to get non-standard symbols from netdb.h
+(and fix a bug in the Makefile when using that). Patching configure
+instead of configure.ac avoids the need for autoconf.
+
+Define __BSD_VISIBLE for feenableexcept and fedisableexcept.
+
+Disable the usage of unsupported GLUT functions.
+
+Change a hard-coded -fPIE to -fPIC.
+
+diff --git a/LspCpp/CMakeLists.txt b/LspCpp/CMakeLists.txt
+index 6b12b04..8e7ebe6 100644
+--- a/LspCpp/CMakeLists.txt
++++ b/LspCpp/CMakeLists.txt
+@@ -162,7 +162,7 @@ function(lspcpp_set_target_options target)
+         set_property(TARGET ${target} PROPERTY CXX_STANDARD 14)
+     endif()
+ 
+-    set_property(TARGET ${target} PROPERTY CXX_EXTENSIONS OFF)
++    set_property(TARGET ${target} PROPERTY CXX_EXTENSIONS ON)
+ 
+     if (CMAKE_GENERATOR MATCHES "Visual Studio.*")
+         lspcpp_set_target_options_with_nuget_pkg(${target} boost 1.76.0.0)
+diff --git a/Makefile.in b/Makefile.in
+index 346ce10..b6b88ca 100644
+--- a/Makefile.in
++++ b/Makefile.in
+@@ -170,7 +170,7 @@ asymptote.so: $(COREFILES:=.pic.o) glew.o
+ 
+ $(LSP_ROOT)/liblspcpp.a:
+ 	$(CMAKE) -B$(LSP_ROOT) -S$(LSP_ROOT) -DCMAKE_CXX_FLAGS=\
+-	"-fPIE @OPTIONS@ @LSP_CXX_BUILD_FLAGS@" \
++	"-fPIC @OPTIONS@ @LSP_CXX_BUILD_FLAGS@" \
+ 	@LSP_CMAKE_OPTIONS@
+ 	$(MAKE) -C $(LSP_ROOT)
+ 
+@@ -209,7 +209,7 @@ GEN_PREPROCESSED_DEPFILE_BASE_ARGS = \
+     $(PYTHON) gen_preprocessed_depfile.py \
+ 	--cxx-compiler="$(CXX)" \
+ 	--additional-raw-arguments="$(OPTS)" \
+-	--cxx-standard=$(subst gnu,,$(subst c++,,$(CXX_STANDARD)))
++	--cxx-standard=$(subst gnu++,,$(subst c++,,$(CXX_STANDARD)))
+ 
+ $(SYMBOL_FILES:=.raw.i): %.raw.i: %.cc gen_preprocessed_depfile.py
+ 	@echo Generating $@ from $<
+diff --git a/configure b/configure
+index dd93b00..938a245 100755
+--- a/configure
++++ b/configure
+@@ -3325,7 +3325,7 @@ then :
+   enableval=$enable_lsp;
+ fi
+ 
+-cxxstd="c++17"
++cxxstd="gnu++17"
+ 
+ CXXFLAGS=$CXXFLAGS" -std=$cxxstd"
+ 
+diff --git a/fpu.h b/fpu.h
+index 7ca8df0..183db6d 100644
+--- a/fpu.h
++++ b/fpu.h
+@@ -6,6 +6,8 @@
+ #define __GNU_VISIBLE 1
+ #endif
+ 
++#define __BSD_VISIBLE 1
```

Comment:
Note: I opened a Haiku ticket for this: https://dev.haiku-os.org/ticket/19534

Reply:
This was merged in the nightly version, but we have to keep this for beta 5 compatibility.